staging/src/k8s.io/apimachinery/pkg/conversion/converter.go

```diff
@@ -442,8 +442,33 @@ func (c *Converter) doConversion(src, dest interface{}, flags FieldMatchingFlags
 	if fn, ok := c.generatedConversionFuncs.untyped[pair]; ok {
 		return fn(src, dest, scope)
 	}
-	// TODO: consider everything past this point deprecated - we want to support only point to point top level
-	// conversions
+
+	if pair.source == pair.dest {
```

Comment:
would it be better or worse to register a generated self-conversion function when we add an object to a scheme in AddKnownTypeWithName, like:

```go
	// if the type implements DeepCopyInto(<obj>), register a self-conversion
	if m := reflect.ValueOf(obj).MethodByName("DeepCopyInto"); !m.IsZero() && m.Type().NumIn() == 1 && m.Type().NumOut() == 0 && m.Type().In(0) == reflect.TypeOf(obj) {
		if err := s.AddGeneratedConversionFunc(obj, obj, func(a, b interface{}, scope conversion.Scope) error {
			// copy a to b
			reflect.ValueOf(a).MethodByName("DeepCopyInto").Call([]reflect.Value{reflect.ValueOf(b)})
			// clear typemeta to match legacy reflective conversion
			b.(Object).GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{})
			return nil
		}); err != nil {
			panic(err)
		}
	}
```

that would remove all reflective conversion from this method, and would let do `SetGroupVersionKind(schema.GroupVersionKind{})` to match legacy behavior

Reply:
great idea - let me try this

Reply:
Done - seems to work great. Thanks a lot for this suggestion - this is much cleaner now.